pkgs/racket-test-core/tests/racket/stream.rktl

```diff
@@ -220,10 +220,175 @@
+        (list a b)))
+
+(test '((1 2))
+      'stream-cons
+      (for/list ([(a b) (stream-cons (values 1 2) #:eager empty-stream)])
+        (list a b)))
+
+(test '((1 2))
+      'stream-cons
+      (for/list ([(a b) (stream-cons #:eager (values 1 2) #:eager empty-stream)])
+        (list a b)))
+
+(test '((1 2) (3 4))
+      'stream
+      (for/list ([(a b) (stream (values 1 2) (values 3 4))])
+        (list a b)))
+
+(test '((1 2) (3 4))
+      'stream*
+      (for/list ([(a b) (stream* (values 1 2) (stream (values 3 4)))])
+        (list a b)))
+
+(test '((0 1) (1 2))
+      'for/stream
+      (for/list ([(a b) (for/stream ([i 2]) (values i (add1 i)))])
+        (list a b)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; testing lazy operation
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; stream-map
+(let ()
+  (define t (stream-cons 0 t))
+  (define s (stream-filter negative? t))
+  (test #t stream? (stream-map add1 s)))
+
+;; stream-filter
+(let ()
+  (define t (stream-cons 0 t))
+  (define s (stream-filter negative? t))
+  (test #t stream? (stream-filter positive? s)))
+
+(let ()
+  (define val #f)
+  (define st
+    (stream-cons 1
+                 (begin
+                   (set! val #t)
+                   empty-stream)))
+
+  (stream-first st)
+  (test #f 'stream-cons val)
+
+  (define st* (stream-filter (lambda (x) #t) st))
+  (stream-first st*)
+  (test #f 'stream-filter val))
+
+
+;; stream-take
+(let ()
+  (define t (stream-cons 0 t))
+  (define s (stream-filter negative? t))
+  (test #t stream? (stream-take s 10)))
+
+;; stream-append
+(let ()
+  (define t (stream-cons 0 t))
+  (define s (stream-filter negative? t))
+  (test #t stream? (stream-append s s)))
+
+;; stream-add-between
+(let ()
+  (define t (stream-cons 0 t))
+  (define s (stream-filter negative? t))
+  (test #t stream? (stream-add-between s 1)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; testing memoizing operation
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; stream-map
+(let ()
+  (define acc 0)
+  (define (f x y)
+    (set! acc (add1 acc))
+    (values (+ 1 x) (+ 2 y)))
+  (define t (stream-cons (values 0 1) t))
+  (define s (stream-map f t))
+  (test '(1 3) call-with-values (λ () (stream-first s)) list)
+  (test '(1 3) call-with-values (λ () (stream-first s)) list)
+  (test 1 'stream-map acc)
+  (test '(1 3) call-with-values (λ () (stream-first (stream-rest s))) list)
+  (test '(1 3) call-with-values (λ () (stream-first (stream-rest s))) list)
+  (test 2 'stream-map acc))
+
+;; stream-filter
+(define-syntax-rule (terminate-quickly e)
+  (let ()
+    (define-values (xs cpu real gc) (time-apply (λ () e) '()))
+    (when (run-unreliable-tests? 'timing)
+      (test #t < real 500))
+    (apply values xs)))
+
+(let ()
+  (define st
+    (for/stream ([i 10000001])
+      (modulo i 10000000)))
```

Comment:
This test takes a while — as intended, but it's slower than I would normally include in this test suite. Would the test still work about as well with one less zero?

Reply:
On my machine (M1), it takes 2.2s.

What about changing the number to `2000000`, which would reduce the execution time to ~0.5s? And we then adjust `terminate-quickly` to check that `real` is less than `100`.

Reply:
> On my machine (M1), it takes 2.2s.

I bet it takes 21s on your machine if you run BC. Those times really are slower than I would normally include in this test suite — especially when the result is only used when opting into unreliable timing tests (so the work usually has no benefit).

If the test needs to be so large to be effective, then maybe it should be in "racket-test/tests/racket"? (It doesn't seem so large/slow that is need to be in the "stress" subdirectory, though.)